src/main/java/ch/njol/skript/conditions/CondStartsEndsWith.java

```diff
@@ -63,15 +63,38 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	
 	@Override
 	public boolean check(Event e) {
-		String affix = this.affix.getSingle(e);
-		
-		if (affix == null) {
-			return false;
-		}
-		
+		String[] affixes = this.affix.getAll(e);
+		if (affixes.length < 1) return false;
 		return strings.check(e,
-				string -> usingEnds ? string.endsWith(affix) : string.startsWith(affix),
-				isNegated());
+			string -> {
+				if (usingEnds) {
+					if (this.affix.getAnd()) {
+						for (String str : affixes) {
+							if (!string.endsWith(str))
+								return false;
+						}
+						return true;
+					} else {
+						for (String str : affixes) {
+							if (string.endsWith((str)))
+								return true;
+						}
+					}
+				} else if (this.affix.getAnd()) {
+					for (String str : affixes) {
+						if (!string.startsWith(str))
+							return false;
+					}
+					return true;
+				} else {
+					for (String str : affixes) {
+						if (string.startsWith((str)))
+							return true;
+					}
+				}
```

Comment:
I think this will improve readability, but it's open to discussion.
```suggestion
				} else { // Using 'starts with'
					if (this.affix.getAnd()) {
						for (String str : affixes) {
							if (!string.startsWith(str))
								return false;
						}
						return true;
					} else {
						for (String str : affixes) {
							if (string.startsWith((str)))
								return true;
						}
					}
				}
```

Reply:
Yeah